server/src/server.rs

```diff
@@ -0,0 +1,465 @@
+use std::{
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
+
+#[cfg(feature = "vllm")]
+use atoma_backends::{GenerateRequest, ServiceRequest};
+use axum::{
+    extract::State,
+    http::{header, HeaderMap, StatusCode},
+    response::{sse::KeepAlive, IntoResponse, Sse},
+    routing::post,
+    Json, Router,
+};
+use serde_json::{json, Value};
+use tokio::{
+    net::TcpListener,
+    signal,
+    sync::{
+        mpsc::{self, UnboundedSender},
+        oneshot,
+    },
+    task::JoinHandle,
+};
+use tracing::{error, info, instrument};
+use utoipa::{OpenApi, ToSchema};
+use utoipa_swagger_ui::SwaggerUi;
+
+use crate::{
+    api::{
+        chat_completions::{ChatCompletionResponse, RequestBody},
+        validate_schema::validate_with_schema,
+    },
+    stream::Streamer,
+};
+
+pub const CHAT_COMPLETIONS_PATH: &str = "/chat/completions";
```

Comment:
There's a doc comment for this that was left behind in the diff

Reply:
Addressed